include/saber/utility.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef SABER_UTILITY_HPP
+#define SABER_UTILITY_HPP
+#pragma once
+
+namespace saber {
+
+namespace detail {
+
+// "primary" template class declaration (2 type args)
+template<typename ToType, typename FromType>
+struct ConvertTraits
+{
+    // Return a |ToType| as converted from |FromType|
+	ToType operator()(const FromType& inValue);
```

Comment:
Subjective:  reference operator: `operator()(...)` is technically `const`...
Since it will not modify `this`.
Consider declaring it: `const`